loky/backend/popen_loky.py

```diff
@@ -195,13 +196,17 @@ def thread_is_spawning():
         semaphore_tracker._semaphore_tracker._fd = args.semaphore
 
     exitcode = 1
+
     try:
-        from_parent = os.fdopen(r, 'rb')
-        prep_data = pickle.load(from_parent)
-        spawn.prepare(prep_data)
-        process_obj = pickle.load(from_parent)
-        from_parent.close()
-        from_parent = None
+        with os.fdopen(r, 'rb') as from_parent:
+            process.current_process()._inheriting = True
+            try:
+                prep_data = pickle.load(from_parent)
+                spawn.prepare(prep_data)
+                process_obj = pickle.load(from_parent)
+            finally:
+                del process.current_process()._inheriting
```

Comment:
Fix the `_check_not_importing_main` function. It was previously broken and should be enforce correctly to avoid any infinite spawning for unprotected scripts.